src/galileo/middleware/tracing.py

```diff
@@ -0,0 +1,204 @@
+
+        # Store in context variables (thread-safe for async)
+        trace_id_token = _trace_id_context.set(trace_id)
+        parent_id_token = _parent_id_context.set(parent_id)
+
+        try:
+            # Process the request
+            return await call_next(request)
+        finally:
+            # Clean up context variables
+            _trace_id_context.reset(trace_id_token)
+            _parent_id_context.reset(parent_id_token)
+
+
+def get_request_logger(project: Optional[str] = None, log_stream: Optional[str] = None) -> GalileoLogger:
+    """
+    Get a request-scoped GalileoLogger configured with distributed tracing context.
+
+    This function should be called within a request handler after the TracingMiddleware has
+    been registered. It creates a new GalileoLogger instance per request that automatically
+    continues the distributed trace from the upstream service.
+
+    If no tracing headers were present in the request, a regular logger is returned.
+
+    Note: This creates a new logger per request, unlike the decorator's get_logger_instance()
+    which uses a singleton pattern.
+
+    Parameters
+    ----------
+    project : Optional[str]
+        Project name (defaults to GALILEO_PROJECT env var)
+    log_stream : Optional[str]
+        Log stream name (defaults to GALILEO_LOG_STREAM env var)
+
+    Returns
+    -------
+    GalileoLogger
+        A logger instance configured for the current request's trace context
+
+    Examples
+    --------
+    ```python
+    @app.post("/process")
+    async def process_request(data: dict):
+        logger = get_request_logger()
+
+        # This span will be attached to the distributed trace
+        logger.add_workflow_span(input=str(data), name="process_workflow")
+        result = await process(data)
+        logger.conclude(output=str(result))
+
+        logger.flush()
+        logger.terminate()
+
+        return {"result": result}
+    ```
+
+    ```python
+    @app.post("/retrieve")
+    async def retrieve_endpoint(query: str):
+        # Get logger with trace context from upstream service
+        logger = get_request_logger()
+
+        # If trace context exists, this creates a workflow span
+        # Otherwise, it starts a new trace
+        if logger.trace_id:
+            logger.add_workflow_span(input=query, name="retrieval_service")
+        else:
+            logger.start_trace(input=query, name="retrieval_service")
+
+        results = retrieve(query, logger)
+
+        logger.conclude(output=str(results))
+        logger.flush()
+        logger.terminate()
+
+        return {"results": results}
+    ```
+    """
+    # Get trace context from middleware
+    trace_id = _trace_id_context.get()
+    parent_id = _parent_id_context.get()
+
+    # Create logger with trace context
+    return GalileoLogger(
+        project=project, log_stream=log_stream, experimental={"mode": "streaming"}, trace_id=trace_id, span_id=parent_id
```

Comment:
Can the `span_id` be set to a `trace_id`? I think you're setting both `trace_id` and `parent_id` to the `trace_id` if there is no parent span